### ISSUE TITLE
[1.11.2] Client Crash while viewing manual

### ISSUE THREAD

**Issue by kreezxil** [BluSunrize/ImmersiveEngineering#2225]
I'm getting client crashes but no crash report is being generated nor is there one in the latest.log. It's like immersive engineering has a try/catch that prevents crash output.

What is happening is that randomly while viewing the immersive engineering manual the client will crash. I also have JEI and I'm using 1.11.2 on forge build 2345.

I am the developer of the World of Dragons modpack. the current modlist is: https://gist.github.com/23c40615aae275bcc453fc0219ee6138

Is there anything I can activate in a config to get helpful information to attach to this?

**Comment by malte0811** [comment]
If you actually crash there will be SOMETHING at the end of the log, even though it may not be the typical MC crash report. Could you post the last ~200 lines of the log? Also what Java version are you using for MC?

**Comment by Alaberti** [comment]
I'm getting a crash on page 4 when trying to view the heavy shield: https://pastebin.com/fSiARdHx

This is on 1.12 (v67) - but I suspect it may be related to this issue, despite the version differences.

**Comment by BluSunrize** [comment]
How could you possibly assume them to be related? THey have nothing in common.

What did you set the keybinding too to equip IE's shield with the magnetic glove? Check your options, provide a screenshot if possible.

**Comment by Alaberti** [comment]
https://cdn.discordapp.com/attachments/190498382085619712/338432663524343810/Screen_Shot_2017-07-22_at_17.30.23.jpg

**Comment by BluSunrize** [comment]
Funky, guess it doesn't like translating mouse buttons atm.
I'll look into fixing that at some point

**Comment by kwpugh** [comment]
I have some players that are getting a client crash while reading the manual, but not everyone.

The client doesn't generate a crash log, but generates this: https://gist.github.com/anonymous/7c77f51914d1a6b2f7156ea1d4ca27dd

**Comment by malte0811** [comment]
They should try updating java, the version that comes with MC is ~3 years old IIRC.

**Comment by banpower1** [comment]
Hey, i updated a modpack i'm playing on today, and started getting crashes if i view the manual for too long, the console doesn't show anything out of the ordinary except for saving the worlds, and two files appeared in the root of the modpack folder:
hs_err_pid3136.log  - https://gist.github.com/banpower1/73b78848c2c0931a2369faa5e594586c
replay_pid14084.log - https://gist.github.com/banpower1/6c65c7ca03712831fdbe5cbbb8cfb50b

The version used is: ImmersiveEngineering-0.10-61
The modpack is Age of Engineering v 1.1.1b (immersive engineering comes with this modpack.

Steps to recreate: Enter world, open 

The last couple of lines, just to 100% verify that they aren't helping anyone:
```
21:30:22 | net.minecraft.client.gui.GuiNewChat | [CHAT] <§2Banpower1§r> Hello from the client, marking fully loaded no fps stutters
21:30:41 | net.minecraft.server.MinecraftServer | <§2Banpower1§r> opening manual
21:30:41 | net.minecraft.client.gui.GuiNewChat | [CHAT] <§2Banpower1§r> opening manual
```
And here is a screenshot of the console (and yes it is 100% scrolled to the very bottom) to show that there is nothing: [https://imgur.com/VmLLNvG](https://imgur.com/VmLLNvG)

**Comment by malte0811** [comment]
Read my comment right above your one.

**Comment by banpower1** [comment]
@malte0811 i just updated my java-64bit to version 144, which (as of writing) is the latest on the java.com site. The crash happened again with the same two error reports, so updating java does not seem to fix the issue

**Comment by mindforger** [comment]
@banpower1 if the log you uploaded is the latest one, you still are using the built-in JRE

> JRE version: Java(TM) SE Runtime Environment (8.0_25-b18) (build 1.8.0_25-b18)

maybe you should check your launcher config

**Comment by banpower1** [comment]
So the new native launcher uses its own java install to launch from, instead of attempting to go with the already installed version on the system. Going through the native launcher and forcing it to go with my newly installed version 144 seems to have fixed the issue (haven't crashed after 5 minutes, when i would normally crash within max 2).
For anyone using the **Twitch Launcher** coming here later, beware that the native launcher doesn't give a danm what your twitch launcher thinks, so you have to manually config the modpack in the native launcher (hit the burger menu on the top right, then Launch options).

**TL;DR**: Native launcher uses own install of java, so "updating java" is too vague troubleshoot advice in the future

`</issue>`

**Comment by mindforger** [comment]
strangely i set it up in the native launcher once 

and now all my twitch launcher modpacks actually use the system JRE instead of the built in without issues

**Comment by Eldresh** [comment]
Myself and everyone on my server are getting the same problem on 1.12.1 using Forge version 14.22.0.2475. The log files it produces are below.

[hs_err_pid12672.txt](https://github.com/BluSunrize/ImmersiveEngineering/files/1319756/hs_err_pid12672.txt)

[replay_pid12672.txt](https://github.com/BluSunrize/ImmersiveEngineering/files/1319761/replay_pid12672.txt)


This is easily replicated on our ends by simply leaving the manual open for a while, it should crash some time within a minute or two (at the most).

**Comment by Johannason** [comment]
Tried to switch Minecraft to use my installed Java version rather than the bundled version. Perhaps I didn't do that right, just pointed it to use javaw.exe in my Program Files/Java since it's using javaw.exe in the Minecraft subfolder by default, figured that was logical.
It promptly exploded.
"Unable to open virtual environment", and that's if it provides an error at all. Sometimes it just fails and returns to the launcher, providing no log of any kind, and producing no game output.

**Comment by Eldresh** [comment]
Same thing happens to me when I tried to use my system's version of Java (an up to date one and one only slightly more up to date than the bundled one).

**Comment by Fluerspar** [comment]
Just thought I'd mention this with my own testing, when you alter your java exec, the installation needs to be in the same driver as your game installation. Example, originally my java was installed on my C: Driver had to reinstall my java to my D: Driver where all my game installations are. Javaw.exe is the correct executable by the way.

**Comment by mindforger** [comment]
Drive not Driver ... dunno if that was jsut a repeated typo ... normally you configure the "bin" folder of the JVM installation and then the launcher should grab the right file itself

**Comment by BluSunrize** [comment]
Even if you set it manually, it's javaw, because that one doesn't open a console window. =P

**Comment by Johannason** [comment]
@Fluerspar It is on the same drive. I just tried copy/pasting my JRE into the Minecraft installation folder, alongside the bundled Java, and pointing to it from there.
"Unable to create Java virtual environment" again. No indication as to why.

**Comment by Fluerspar** [comment]
@mindforger Yeah, I noticed that this morning lmao, but thank you for pointing that out, need to add that to my auto correct app. @Johannason did you remove the original "bin" folder that was in the minecraft installation? Replace that and see if it works that way. If you haven't already that is. I'll see how it works out on my end and give you another reply.

**Comment by Fluerspar** [comment]
I'm an idiot, okay what I worked with I'm not exceptionally proficient with everything but I just replaced the java that is in Minecraft's installation folder with Java Version 144 x64 bit and it launched just fine.
@johannason could try that just make copies of the original java in minecrafts installation and replace all the files with an updated java it should work. If not then I've no idea what error is on your end of things.
@mindforger Thanks for that had to add that word to my auto correct app, kept correcting to that without me noticing lmao.

**Comment by Eldresh** [comment]
Figured out the problem on my end (as far as updating Java was concerned). I was using the 32 bit version and not the 64 bit one (derp). Updating it seems to have solved the issue completely.

**Comment by Johannason** [comment]
@Fluerspar Updating to Java x64 has stopped the book crash. I had to explicitly dig for the x64 install, since the one Java gave me as "Recommended" was x32, and would throw "Unable to create virtual environment".

**Comment by mindforger** [comment]
your Browser is running 32bit that is why the Website recommends you the wrong version

**Comment by KocLobster** [comment]
Wanted to take a second to thank everyone in this thread. Until just now, I had no idea that Curse was using a 3 year old version of Java (1.8.0_25). How stupid can you be Curse? I have always kept my Java up to date for years now, so this was surprised. Told the Curse client to stop using the native launcher, and now I'm running with the most recent version of Java.

Sure enough, no longer using the native launcher and ensuring that MC is running on Java144 solved my issue, I had been dealing with the IE manual randomly crashing for quite some time. I totally love the way this manual works, so it was really frustrating.

Thanks for making a great mod and donating your time to the MC community @BluSunrize

**Comment by mindforger** [comment]
>How stupid can you be Curse?

Even better, it is minecraft himself shipping this JRE

**Comment by uecasm** [comment]
It's not really a good solution to just say "use a newer Java", especially when most of the launchers will force you back to the original MC java version on every launch, ignoring settings.  Ideally it should be made to work with the MC-supplied java.

**Comment by malte0811** [comment]
It is not a bug on our side, it is a bug in the JRE as far as we know. We don't know what causes it, finding out what causes it would take forever and probably result in knowing what causes it and knowing we can't fix it. Aside from that using up-to-date Java is something people should do regardless of specific bugs.

**Comment by jikuja** [comment]
> We don't know what causes it, finding out what causes it would take forever and probably result in knowing what causes it and knowing we can't fix it.

I don't know if you used time with crashlogs but for me it looks like JVM's JIT compiler crashes with blusunrize.lib.manual.gui.GuiManual::func_73863_a

But true, it might be hard to fix unless JVM changelogs mention what kind of code triggers the crash

**Comment by malte0811** [comment]
I know that, I'm quite sure I even found the bug report on the JVM for this bug. The problem is: that report is written for people who know how the JIT and compilers in general work. I don't. So I have no idea what they are talking about and how I could fix it.
